twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -846,24 +875,31 @@ private boolean handleRestartRunnablesInstances(final Message message, final Run
   /**
    * Helper method to restart instances of runnables.
    */
-  private void restartRunnableInstances(String runnableName, @Nullable Set<Integer> instanceIds) {
+  private void restartRunnableInstances(String runnableName, @Nullable Set<Integer> instanceIds)
+    throws InterruptedException {
```

Comment:
I don't think this method throws `InterruptedException`. This also mean we don't have to change the `handleRestartRunnablesInstances` method.